internal/client/gcp.go

```diff
@@ -0,0 +1,247 @@
+			garmPoolID:       spec.BootstrapParams.PoolID,
+			garmControllerID: spec.ControllerID,
+			osType:           string(spec.BootstrapParams.OSType),
+		},
+	}
+
+	insertReq := &computepb.InsertInstanceRequest{
+		Project:          g.cfg.ProjectId,
+		Zone:             g.cfg.Zone,
+		InstanceResource: inst,
+	}
+
+	op, err := g.client.Insert(ctx, insertReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create instance %s: %w", insertReq, err)
+	}
+
+	if err = op.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("failed to wait for operation: %w", err)
+	}
+
+	return inst, nil
+}
+
+func (g *GcpCli) GetInstance(ctx context.Context, instanceName string) (*computepb.Instance, error) {
+	req := &computepb.GetInstanceRequest{
+		Project:  g.cfg.ProjectId,
+		Zone:     g.cfg.Zone,
+		Instance: instanceName,
+	}
+
+	instance, err := g.client.Get(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instance: %v", err)
+	}
+
+	return instance, nil
+}
+
+func (g *GcpCli) ListDescribedInstances(ctx context.Context, poolID string) ([]*computepb.Instance, error) {
+	label := fmt.Sprintf("labels.garmpoolid=%s", poolID)
+	req := &computepb.ListInstancesRequest{
+		Project: g.cfg.ProjectId,
+		Zone:    g.cfg.Zone,
+		Filter:  &label,
+	}
+
+	it := g.client.List(ctx, req)
+	var instances []*computepb.Instance
+	for {
+		instance, _ := it.Next()
+		if instance == nil {
+			break
+		}
+		instances = append(instances, instance)
+	}
+
+	return instances, nil
+}
+
+func (g *GcpCli) DeleteInstance(ctx context.Context, instance string) error {
+	req := &computepb.DeleteInstanceRequest{
+		Instance: util.GetInstanceName(instance),
+		Project:  g.cfg.ProjectId,
+		Zone:     g.cfg.Zone,
+	}
+
+	op, err := g.client.Delete(ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to delete instance: %w", err)
+	}
+
+	if err = op.Wait(ctx); err != nil {
+		return fmt.Errorf("unable to wait for the delete operation: %w", err)
+	}
+
+	return nil
+}
+
+func (g *GcpCli) StopInstance(ctx context.Context, instance string) error {
+	req := &computepb.StopInstanceRequest{
+		Instance: util.GetInstanceName(instance),
+		Project:  g.cfg.ProjectId,
+		Zone:     g.cfg.Zone,
+	}
+
+	op, err := g.client.Stop(ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to stop instance: %w", err)
+	}
+
+	if err = op.Wait(ctx); err != nil {
+		return fmt.Errorf("unable to wait for the operation: %w", err)
+	}
+
+	return nil
+}
+
+func (g *GcpCli) StartInstance(ctx context.Context, instance string) error {
+	req := &computepb.StartInstanceRequest{
+		Instance: util.GetInstanceName(instance),
+		Project:  g.cfg.ProjectId,
+		Zone:     g.cfg.Zone,
+	}
+
+	op, err := g.client.Start(ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to start instance: %w", err)
+	}
+
+	if err = op.Wait(ctx); err != nil {
+		return fmt.Errorf("unable to wait for the operation: %w", err)
+	}
+
+	return nil
+}
+
+func selectStartupScript(osType string) string {
```

Comment:
you can use `params.OSType` (https://github.com/cloudbase/garm-provider-common/blob/main/params/params.go#L28-L32) here, and skip casting to string.